circus-train-core/src/main/java/com/hotels/bdp/circustrain/core/replica/Replica.java

```diff
@@ -445,4 +437,23 @@ public TableAndStatistics getTableAndStatistics(TableReplication tableReplicatio
         tableReplication.getReplicaTableName());
   }
 
+  public void checkIfReplicaCleanupRequired(
```

Comment:
This does more than just check, it also performs the cleanup. Perhaps rename to `cleanupReplicaIfRequired`. Or maybe better move the `if (replicationMode == FULL_OVERWRITE) {` up to where it was before and then call this method `cleanupReplica` as it only applies to that mode.